shuttle/src/runtime/task/mod.rs

```diff
@@ -664,6 +793,19 @@ impl Task {
             }
         )
     }
+
+    /// Get the next_event with a downcast lifetime tied to self.
+    pub(crate) fn next_event(&self) -> &Event<'_> {
+        unsafe { std::mem::transmute(&self.next_event) }
+    }
+
+    pub(crate) unsafe fn set_next_event(&mut self, event: Event<'_>) {
```

Comment:
SAFETY comment describing how to make it safe. Also describe why unsafe is needed